### PR TITLE
Update CI/CD actions, use UV_ENV, use OpenID Connect to pypi

### PR DESCRIPTION
**Pull request by lukaspie**

### CODE REVIEW

**Review by domna — APPROVED** [review]

**Comment by domna** [comment]
You can also remove the PYPI_API_TOKEN from the secrets in the settings of the repo then.

File: .github/workflows/publish.yml
Comment (domna) [inline .github/workflows/publish.yml:1]: Ah sorry. I already approved aber you need 
```
permissions:
      id-token: write
```

see https://github.com/FAIRmat-NFDI/pynxtools-mpes/blob/main/.github/workflows/publish.yml#L14

Comment (lukaspie) [reply .github/workflows/publish.yml:1]: Right, thanks for noticing. I updated the publish workflow, could you have another look?

**Review by domna — APPROVED** [review]

File: .github/workflows/publish.yml
Comment (domna) [reply .github/workflows/publish.yml:1]: Yes looks good now

**Pull request merged**